Submitted By: Pierre Labastie <pierre AT higgs DOT linuxfromscratch DOT org>
Date: 2014-04-12
Initial Package Version: 2.4.6
Upstream Status: From upstream Node IDe8e59c50a451b45bd10cd495b81e5c61b2b4bca8
Origin: upstream
Description: PR1726: configure fails looking for ecj.jar before even trying to find javac


diff -r 46f9059509fe -r e8e59c50a451 acinclude.m4
--- a/acinclude.m4	Fri Mar 28 18:06:31 2014 +0000
+++ b/acinclude.m4	Fri Mar 28 18:12:46 2014 +0000
@@ -134,6 +134,12 @@
   IT_USING_ECJ
   IT_WITH_GCJ
   
+  if test "x${ECJ_JAR}" = "xno"; then
+    if test "x${JAVAC}" = "x"; then
+      AC_MSG_ERROR("No compiler or ecj JAR file was found.")
+    fi
+  fi
+
   if test x"${GCJ}" != xno ; then
     JAVAC="${JAVA} -classpath ${ECJ_JAR} org.eclipse.jdt.internal.compiler.batch.Main"
   fi
@@ -350,11 +356,6 @@
       fi
   fi
   AC_MSG_RESULT(${ECJ_JAR})
-  if test "x${ECJ_JAR}" = "xno"; then
-    if test "x${JAVAC}" = "x"; then
-      AC_MSG_ERROR("No compiler or ecj JAR file was found.")
-    fi
-  fi
   AC_SUBST(ECJ_JAR)
 ])
 

